### ISSUE TITLE
Looking for packagers

### ISSUE THREAD

**Issue by Dj-Codeman** [Dj-Codeman/doge#35]
I have package in the AUR for dns-doge (I broke it and haven't gotten to fixing it). I'm looking for people interested in packaging doge and educating me about it. I don't have any distros in mind now so this is kinda a discussion place for now

**Comment by aktaboot** [comment]
I packaged it on nixpkgs, 

as for AUR, I never really got into their build system
But In the current AUR PKGBUILD, you'd need to add openssl as a dependency and cargo as a 'makedepends' dependency (i.e build-time dependency) for starters :)

**Comment by CarterLi** [comment]
Requested for packaging on homebrew. You may look at the configuration file of dog

https://github.com/Homebrew/homebrew-core/blob/master/Formula/d/dog.rb

**Comment by Dj-Codeman** [comment]
@aktaboot THX ! I've been meaning to try nix def will give it a shot. As for aur I have it packaged already, It may need an update but I'm slowly learning with that.

**Comment by brian6932** [comment]
I'm packaging for scoop, and I need you to rerun this job https://github.com/Dj-Codeman/doge/actions/runs/10725489969, there's currently no bin for the latest `0.2.7` pre-release.

**Comment by Dj-Codeman** [comment]
@brian6932 I'll do you one better. Just knocked out a few small things and the workflow file and released 0.2.8 proper with the binaries !!

**Comment by brian6932** [comment]
https://github.com/Dj-Codeman/doge/releases/v0.2.8 is missing artifacts from the https://github.com/Dj-Codeman/doge/actions/runs/12155914713 workflow.

**Comment by moonfruit** [comment]
This project is not up to the condition of being accepted by homebrew for now, so I added it to my own [homebrew-tap](https://github.com/moonfruit/homebrew-tap).
You might want to consider creating your own homebrew tap as well. @Dj-Codeman 

[Homebrew Acceptable Formulae](https://docs.brew.sh/Acceptable-Formulae).

**Comment by CarterLi** [comment]
@brian6932 Does doge work for you on Windows? It keeps reporting `Error [network]: A socket operation was attempted to an unreachable network (os error 10051)` to me

**Comment by brian6932** [comment]
Works for me, although I built it myself, didn't try the prebuilt bins.

**Comment by Dj-Codeman** [comment]
> This project is not up to the condition of being accepted by homebrew for now, so I added it to my own [homebrew-tap](https://github.com/moonfruit/homebrew-tap). You might want to consider creating your own homebrew tap as well. [@Dj-Codeman](https://github.com/Dj-Codeman)
> 
> [Homebrew Acceptable Formulae](https://docs.brew.sh/Acceptable-Formulae).

It's the whole being a fork and not adopted by any distros as a replacement section I assume ?

**Comment by Dj-Codeman** [comment]
> [@brian6932](https://github.com/brian6932) Does doge work for you on Windows? It keeps reporting `Error [network]: A socket operation was attempted to an unreachable network (os error 10051)` to me

Was It the 0.2.8 build for the binary ?

**Comment by thomasmerz** [comment]
@roseswe because `dog` is already in **SUSE** repos "openSUSE-Leap-15.6-Oss" can you find someone at SUSE who can look to maintain and package from this forked and active maintained repo?

**Comment by roseswe** [comment]
@thomasmerz - maintain for openLEAP or SLES itself?

**Comment by thomasmerz** [comment]
I have no idea how SUSE's Open Build Service (for openLeap) works. But because `dog` is already in SUSE repo "openSUSE-Leap-15.6-Oss" this package already must have found some way to SUSE 😄 But with a very old version (0.1.0 vs. current 0.2.8) 😢 

So my question to [SUSE](https://github.com/SUSE) is:
If this nice and fancy tool is already in openSUSE-Leap-15.6-Oss repo, who will keep it uptodate, because this is currently not the case? Maybe you can trigger someone?

On [openbuildservice](https://build.opensuse.org/package/show/openSUSE:Leap:15.6:Update/dog) I see that [Yuchen Lin ](https://build.opensuse.org/users/maxlin_factory) worked on this… But he's not a person listed on https://github.com/orgs/SUSE/people

**Comment by ashugg** [comment]
> > This project is not up to the condition of being accepted by homebrew for now, so I added it to my own [homebrew-tap](https://github.com/moonfruit/homebrew-tap). You might want to consider creating your own homebrew tap as well. [@Dj-Codeman](https://github.com/Dj-Codeman)
> > [Homebrew Acceptable Formulae](https://docs.brew.sh/Acceptable-Formulae).
> 
> It's the whole being a fork and not adopted by any distros as a replacement section I assume ?

Not sure, but @moonfruit's formula went into Homebrew main less than a week ago - congrats!

As it's quite new, should it be renamed to dog-community? Looks like "doge" was a placeholder name.

**Comment by Dj-Codeman** [comment]
> > > This project is not up to the condition of being accepted by homebrew for now, so I added it to my own [homebrew-tap](https://github.com/moonfruit/homebrew-tap). You might want to consider creating your own homebrew tap as well. [@Dj-Codeman](https://github.com/Dj-Codeman)
> > > [Homebrew Acceptable Formulae](https://docs.brew.sh/Acceptable-Formulae).
> > 
> > 
> > It's the whole being a fork and not adopted by any distros as a replacement section I assume ?
> 
> Not sure, but [@moonfruit](https://github.com/moonfruit)'s formula went into Homebrew main less than a week ago - congrats!
> 
> As it's quite new, should it be renamed to dog-community? Looks like "doge" was a placeholder name.

It was "doge" for the joke, then doge became .... not a joke. I've been dragging feet about it because I don't know what the bin name would change to. dog_community is fine for the repo but prob not for the bin. Also finally got a macbook so the brew / Homebrew stuff is more immediately relevant. I saw the formula get pushed and choked on my coffee

**Comment by ashugg** [comment]
> It was "doge" for the joke, then doge became .... not a joke. I've been dragging feet about it because I don't know what the bin name would change to. dog_community is fine for the repo but prob not for the bin. Also finally got a macbook so the brew / Homebrew stuff is more immediately relevant. I saw the formula get pushed and choked on my coffee

I agree that `dog-community` or `dog_community` is not going to be something anyone wants to type into their terminal, even with tab-completion. (PowerShell users may be outliers here. I do not understand them, and do not speak for them.)

I'm old enough that `dog` is still stuck in my mind as an alternate to `cat` on GNU/Linux systems in the late 1990s and early 2000s. It was a project by Jason Cohen and Jacob Leverich of photodex.com, which started in 1999 and disappeared off the Internet somewhere in 2007 or 2008. Debian's last release of the package was in March 2008. They kept "dog" in the distribution for a while, until it was removed at some point for being abandoned upstream and enough time had passed with no-one picking up the torch and continuing to maintain the project. (My thanks to the [Debian Archive](https://archive.debian.org/debian/pool/main/d/dog/) for keeping all the old sources and diffs around for me to look at 20 years later!)

For the most part, I think that [bat](https://github.com/sharkdp/bat) has been accepted as the modern replacement for that original "dog", as an alternative to the venerable cat(1) tool. (And, it's my problem that I still associate "dog" with "cat", after all this time.)

The Cargo.toml file makes it easy to change the build artefacts' base name from "doge" to something else, but "doge" is referenced elsewhere in the source tree now and should be cleaned up to consistently use one name.

So, can there be just one tool going forward that's called "dog", that everyone will associate with DNS queries?

I would hope that there is agreement that in the most part, the "dog" that _your_ dog_community project is keeping alive is the successor to Benjamin Sago's [original project](https://github.com/ogham/dog/). Really, it is lacking only official closure/archival of that project and/or a statement from him that blesses dog_community as the official continuation of his work. Then there would be "dog" provided by dog_community, and "doggo" as the "spiritual sibling" [tool written in Go](https://github.com/mr-karan/doggo) that was inspired by Mr Sago's "dog".

(I did look, and if Mr Sago is still maintaining an online presence now in late 2025, he is doing so separately to his previous social media identities. The domains referenced in his "dog" project either are no longer registered in DNS or refer to websites that have been offline since late 2024. I hope he is alright. Maybe he's just sick of the Internet and noped out, which I think is a position we can all relate to, and may have done ourselves at some point.)

I suggest adding a document (separately to README.md) that shows the known history of dog. Briefly, there is the original upstream project (added to GitHub in April 2020, with hints in the code that work began in 2018), its first and last code release in Nov 2020, the creation of Golang clone "doggo" (Dec 2020), the original project going silent after Oct 2021, and then "The dawn of a new dog" when you created this fork in Feb 2024.

Apologies for this being so long. I'm one of those people. 😉

# TL;DR
tl;dr: make the build name (binary, man page etc) default to "dog" but easily configurable to something else like doge/dogrs/dogdns by an end user or downstream package maintainer to suit their whims or policies; add project timeline/history document

**Comment by moonfruit** [comment]
I’m sorry for using `doge` as the name of the Homebrew formula. It’s because it had long existed under that name in my own [tap](https://github.com/moonfruit/homebrew-tap) and could safely coexist with the original `dog` in the PATH. However, `dog` is indeed a better name, and since Homebrew has now removed support for the original `dog`, it might be a good idea to rename `doge`’s binary to `dog`.

**Comment by goto-dev-null** [comment]
Bearing in mind that the discussion above about renaming the binary would mean some additional tweaks, I've got the PKGBUILD updated to work.

Unfortunately, the original [doge-dns](https://aur.archlinux.org/packages/dns-doge) package was submitted as version 2.6 (and then 2.7), not 0.2.6 (and 0.2.7). That means that Pacman will see this newer release as a downgrade for anyone who has it installed already. The good news is that, as the AUR package has 0 votes, I do not think that's going to be a ton of people. 😅

**EDIT**: Noticed that the license is incorrect; "EPL" is Eclipse Public License.

_(You will also want to update the .SRCINFO file in the dns-doge repo)_

Here's the diff:

```patch
diff --git a/PKGBUILD b/PKGBUILD
index 294824b..df02f99 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -1,28 +1,28 @@
 # Maintainer: Darrion Whitfield <dwhitfield.ramfield.net>
 pkgname=dns-doge
-pkgver=2.7
+pkgver=0.2.9
 pkgrel=1
 pkgdesc="A dns dig utility with a colorful output and plenty of features"
 arch=('x86_64' 'i686' 'aarch64' 'armv7h')
-url="https://github.com/Dj-Codeman/doge"
+url="https://github.com/Dj-Codeman/dog_community"
-license=('EPL-2.0')
+license=('EUPL-1.2')
-depends=('rust' 'pandoc')
-source=('https://github.com/Dj-Codeman/doge/archive/refs/tags/v0.2.7.tar.gz')
-sha256sums=('7049e2ccd6907f4f4222b8ea84160d65b57aadbbee9498da353a00c576bc647e')
+makedepends=('rust' 'pandoc')
+source=("${url}/archive/refs/tags/v${pkgver}.tar.gz")
+sha256sums=('21d459f1f88d6a1e001a747b84782f180c01de8f3c39f3a1389c352b2f2edc88')
 
 prepare() {
-    cd "${srcdir}/doge-0.2.7"
+    cd "${srcdir}/dog_community-${pkgver}"
     make test-quick
 }
 
 build() {
-    cd "${srcdir}/doge-0.2.7"
+    cd "${srcdir}/dog_community-${pkgver}"
     make build-release
 }
 
 package() {
-    cd "${srcdir}/doge-0.2.7"
+    cd "${srcdir}/dog_community-${pkgver}"
     mkdir -pv "${pkgdir}/usr/bin/" "${pkgdir}/usr/share/man/man1" "${pkgdir}/usr/share/bash-completion/completions/"
     cp "target/release/doge" "${pkgdir}/usr/bin/doge"
 	chmod +x "${pkgdir}/usr/bin/doge"
     # adding man pages

```

And here's the full content:


```bash
# Maintainer: Darrion Whitfield <dwhitfield.ramfield.net>
pkgname=dns-doge
pkgver=0.2.9
pkgrel=1
pkgdesc="A dns dig utility with a colorful output and plenty of features"
arch=('x86_64' 'i686' 'aarch64' 'armv7h')
url="https://github.com/Dj-Codeman/dog_community"
license=('EUPL-1.2')
makedepends=('rust' 'pandoc')
source=("${url}/archive/refs/tags/v${pkgver}.tar.gz")
sha256sums=('21d459f1f88d6a1e001a747b84782f180c01de8f3c39f3a1389c352b2f2edc88')

prepare() {
    cd "${srcdir}/dog_community-${pkgver}"
    make test-quick
}

build() {
    cd "${srcdir}/dog_community-${pkgver}"
    make build-release
}

package() {
    cd "${srcdir}/dog_community-${pkgver}"
    mkdir -pv "${pkgdir}/usr/bin/" "${pkgdir}/usr/share/man/man1" "${pkgdir}/usr/share/bash-completion/completions/"
    cp "target/release/doge" "${pkgdir}/usr/bin/doge"
	chmod +x "${pkgdir}/usr/bin/doge"
    # adding man pages
    pandoc --standalone -f markdown -t man man/doge.1.md > "${pkgdir}/usr/share/man/man1/doge.1"
}

```
